File: frontend/src/services/teams.ts
import { API_BASE } from './apiConfig';

export type TeamMemberSchedule = {
  id: string;
  day: string;
  timeRange: string;
};

export type TeamMemberCard = {
  id: string;
  name: string;
  role: string;
  color: string | null;
  avatarUrl: string | null;
  totalHours: number;
  schedule: TeamMemberSchedule[];
};

export type TeamGroup = {
  role: string;
  members: TeamMemberCard[];
};

export async function getRoster(date: string): Promise<TeamGroup[]> {
  const res = await fetch(`${API_BASE}/teams?date=${encodeURIComponent(date)}`);

  if (!res.ok) {
    throw new Error('Failed to retrieve roster');
  }

  const data = await res.json();

  return data.groups;
}

File: backend/gatorsched_api/schemas/features/teams.py
from datetime import date

from pydantic import BaseModel


class TeamMemberSchedule(BaseModel):
    id: str
    day: date  # needs to be converted to a string  most likely
    timeRange: str


class TeamMemberCard(BaseModel):
    id: str
    name: str
    role: str
    color: str | None = None
    avatarUrl: str | None = None
    totalHours: float
    schedule: list[TeamMemberSchedule]


class TeamGroup(BaseModel):
    role: str
    members: list[TeamMemberCard]


class TeamsResponse(BaseModel):
    groups: list[TeamGroup]
